_posts/2022-12-31-arrow-adbc.md

```diff
@@ -0,0 +1,222 @@
+2. The query is passed on to the ADBC driver.
+3. The driver translates the query to a database-specific protocol and sends the query to the database.
+4. The database executes the query and returns the result set in a database-specific format, which is ideally Arrow data.
+5. The driver translates the result format into Arrow data if needed.
+
+So either way, the client can use a single API and get Arrow data in all cases.
+
+## In More Detail
+
+As mentioned, ADBC is an API standard.
+You can read the [C/C++ header][adbc.h], [Go interfaces][adbc-go], and the [Java interfaces][adbc-core] for yourself.
+The project has built several libraries around this standard:
+
+- Driver implementations implement the API standard on top of a particular protocol or vendor.
+  See the drivers for [libpq (PostgreSQL)][adbc-libpq] and [JDBC][adbc-jdbc].
+- The driver manager library provides an easy way to use multiple drivers from a single program, and also load drivers at runtime if desired.
+
+This infrastructure is reusable by other drivers/vendors and accessible to other languages.
+For example, the Python bindings shown above are actually generic Python bindings for any ADBC driver, courtesy of the driver manager.
+
+ADBC builds on several existing Arrow projects.
+
+- The [Arrow C Data Interface][c-data-interface] provides a cross-language way to pass Arrow data between ADBC implementations and ADBC users.
+- [Flight SQL][flight-sql] is used to implement an ADBC driver, making Flight SQL accessible to languages like Python.
+- [Nanoarrow][nanoarrow] is used to implement some ADBC drivers, making them lightweight and independent of the core Arrow libraries.
+
+Also, ADBC provides API support for the emerging [Substrait][substrait] specification.
+Applications can pass queries either as strings (of SQL) or binary blobs (of serialized Substrait plans).
+
+## What about {Flight RPC, Flight SQL, JDBC, ODBC, …}?
+
+ADBC fills a specific niche that related projects do not address. It is both:
+
+- **Arrow-native**: ADBC can pass through Arrow data with no overhead thanks to the [C Data Interface][c-data-interface].
+  JDBC is row-oriented, and ODBC has implementation caveats, as discussed, that make it hard to use with Arrow.
+- **Vendor-agnostic**: ADBC drivers can implement the API using any underlying protocol, while Flight SQL requires server-side support that may not be easy to add.
+
+Plotted along these axes, we see that ADBC is both vendor-agnostic and Arrow-native, while Flight SQL is vendor-specific, and JDBC/ODBC are row-oriented:
+
+<figure style="text-align: center;">
+  <img src="{{ site.baseurl }}/img/ADBCQuadrants.svg"
+       alt="ADBC is Arrow-native and database-agnostic, while Flight SQL is Arrow-native and database-specific. JDBC and ODBC are row-oriented and database-agnostic, and wire protocols like Postgres's and SQL Server's are row-oriented and database-specific."
+       width="90%" class="img-responsive">
+</figure>
```

Comment:
Good point. Since we're talking about databases, how about a table instead? 
![image](https://user-images.githubusercontent.com/327919/207962602-98df7b4e-babc-466b-97c6-4a06651363b3.png)